File: app/src/main/java/com/jmalinen/blackjack/ui/components/ActionBar.kt
package com.jmalinen.blackjack.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.jmalinen.blackjack.model.PlayerAction
import com.jmalinen.blackjack.ui.theme.GoldAccent

@Composable
fun ActionBar(
    availableActions: Set<PlayerAction>,
    onHit: () -> Unit,
    onStand: () -> Unit,
    onDouble: () -> Unit,
    onSplit: () -> Unit,
    onSurrender: () -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 4.dp, vertical = 12.dp)
    ) {
        ActionButton("Hit", PlayerAction.HIT in availableActions, onHit, Modifier.weight(1f))
        ActionButton("Stand", PlayerAction.STAND in availableActions, onStand, Modifier.weight(1f))
        ActionButton("Double", PlayerAction.DOUBLE_DOWN in availableActions, onDouble, Modifier.weight(1f))
        ActionButton("Split", PlayerAction.SPLIT in availableActions, onSplit, Modifier.weight(1f))
        ActionButton("Surr.", PlayerAction.SURRENDER in availableActions, onSurrender, Modifier.weight(1f))
    }
}

@Composable
fun InsuranceBar(
    availableActions: Set<PlayerAction>,
    onInsurance: () -> Unit,
    onDecline: () -> Unit,
    onEvenMoney: () -> Unit,
    onDeclineEvenMoney: () -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 12.dp),
        horizontalArrangement = Arrangement.SpaceEvenly
    ) {
        if (PlayerAction.EVEN_MONEY in availableActions) {
            ActionButton("Even Money", true, onEvenMoney)
            ActionButton("Decline", true, onDeclineEvenMoney)
        } else {
            ActionButton("Insurance", PlayerAction.INSURANCE in availableActions, onInsurance)
            ActionButton("No Thanks", PlayerAction.DECLINE_INSURANCE in availableActions, onDecline)
        }
    }
}

@Composable
private fun ActionButton(
    text: String,
    enabled: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Button(
        onClick = onClick,
        enabled = enabled,
        shape = RoundedCornerShape(8.dp),
        colors = ButtonDefaults.buttonColors(
            containerColor = GoldAccent,
            contentColor = Color.Black,
            disabledContainerColor = Color.Gray.copy(alpha = 0.3f),
            disabledContentColor = Color.White.copy(alpha = 0.4f)
        ),
        contentPadding = PaddingValues(horizontal = 4.dp, vertical = 8.dp),
        modifier = modifier
            .padding(horizontal = 2.dp)
            .defaultMinSize(minWidth = 1.dp, minHeight = 36.dp)
    ) {
        Text(
            text = text,
            fontWeight = FontWeight.Bold,
            fontSize = 13.sp,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis
        )
    }
}

File: ios/Blackjack/Blackjack/Views/Components/ActionBar.swift
import SwiftUI

struct ActionBar: View {
    let availableActions: Set<PlayerAction>
    let onHit: () -> Void
    let onStand: () -> Void
    let onDouble: () -> Void
    let onSplit: () -> Void
    let onSurrender: () -> Void

    var body: some View {
        HStack(spacing: 4) {
            ActionButton(text: "Hit", enabled: availableActions.contains(.hit), action: onHit)
            ActionButton(text: "Stand", enabled: availableActions.contains(.stand), action: onStand)
            ActionButton(text: "Double", enabled: availableActions.contains(.doubleDown), action: onDouble)
            ActionButton(text: "Split", enabled: availableActions.contains(.split), action: onSplit)
            ActionButton(text: "Surr.", enabled: availableActions.contains(.surrender), action: onSurrender)
        }
        .padding(.horizontal, 4)
        .padding(.vertical, 12)
    }
}

struct InsuranceBar: View {
    let availableActions: Set<PlayerAction>
    let onInsurance: () -> Void
    let onDecline: () -> Void
    let onEvenMoney: () -> Void
    let onDeclineEvenMoney: () -> Void

    var body: some View {
        HStack(spacing: 16) {
            if availableActions.contains(.evenMoney) {
                ActionButton(text: "Even Money", enabled: true, action: onEvenMoney)
                ActionButton(text: "Decline", enabled: true, action: onDeclineEvenMoney)
            } else {
                ActionButton(text: "Insurance", enabled: availableActions.contains(.insurance), action: onInsurance)
                ActionButton(text: "No Thanks", enabled: availableActions.contains(.declineInsurance), action: onDecline)
            }
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
    }
}

private struct ActionButton: View {
    let text: String
    let enabled: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(text)
                .font(.system(size: 13, weight: .bold))
                .lineLimit(1)
                .minimumScaleFactor(0.7)
                .frame(maxWidth: .infinity)
                .padding(.horizontal, 4)
                .padding(.vertical, 8)
        }
        .disabled(!enabled)
        .background(enabled ? Color.goldAccent : Color.gray.opacity(0.3))
        .foregroundColor(enabled ? .black : .white.opacity(0.4))
        .cornerRadius(8)
    }
}
